app/main/controllers/activity/RTMediaBuddyPressActivity.php

```diff
@@ -60,6 +62,60 @@ function __construct() {
 		}
 	}
 
+	/**
+	 * For adding secondary avatar in the activity header.
+	 *
+	 * @param String $action   Has the markup for activity header.
+	 * @param array  $activity Contains values realated to the activity.
+	 *
+	 * @return String $action.
+	 */
+	function bp_get_activity_action_pre_meta( $action, $activity ) {
+
+		if ( 'rtmedia_update' === $activity->type && 'groups' === $activity->component ) {
+
+			switch ( $activity->component ) {
+				case 'groups':
+				case 'friends':
+					// Only insert avatar if one exists.
+					$secondary_avatar = bp_get_activity_secondary_avatar();
+					if ( ! empty( $secondary_avatar ) ) {
+
+						if ( false === strpos( $activity->action, $secondary_avatar ) ) {
+							$reverse_content = strrev( $activity->action );
+							$position        = strpos( $reverse_content, 'a<' );
+							$action          = substr_replace( $activity->action, $secondary_avatar, -$position - 2, 0 );
+						}
+					}
+					break;
+			}
+
+			return $action;
+
+		} else {
+
+			switch ( $activity->component ) {
+				case 'groups':
+				case 'friends':
+					$secondary_avatar = bp_get_activity_secondary_avatar( array( 'linked' => false ) );
+
+					// Only insert avatar if one exists.
+					if ( ! empty( $secondary_avatar ) ) {
+
+						if ( false === strpos( $activity->action, $secondary_avatar ) ) {
```

Comment:
Is it possible to combine these conditions?
`if ( ! empty( $secondary_avatar ) )` and `if ( false === strpos( $activity->action, $secondary_avatar ) )?`

Reply:
Done :+1: